test/parallel/test-http-client-abort-destroy.js

```diff
@@ -58,8 +58,7 @@ const assert = require('assert');
   server.listen(0, common.mustCall(() => {
     const options = { port: server.address().port };
     const req = http.get(options, common.mustNotCall());
-    req.on('error', common.mustCall((err) => {
-      assert.strictEqual(err.code, 'ECONNRESET');
+    req.on('close', common.mustCall((err) => {
```

Comment:
I think this change is not correct.

A user is currently adding `req.on('error')` and `req.on('response')`. They are not adding `req.on('close')`.

Reply:
If that's considered a problem then I think @lpinca might be correct on https://github.com/nodejs/node/pull/20077.

I think the discussion there went from here https://github.com/nodejs/node/pull/20077#issuecomment-440686774

Reply:
This confuses me a little though, `abort()` is what most people use and then  according to your comment above they are already broken.

I think we have to options here:
- Make `destroy()` not force an error.
- Make `abort()` force an error.

Having the two of them behave differently is confusing.

Reply:
Just to clarify, this is not about swallowing errors. This is about forcing an error.

Reply:
The current behavior of `http.request()` implies that either `'error'`  is fired or `'response'` is fired. Firing _nothing_ wil definitely create a memory leak.

Reply:
Then `abort()` has been broken for a long time, because you can call `abort()` and get neither `'error'` nor `'response'`.

This happens if `abort()` is called before the request has received a `socket`.

Should I change this PR to fix that instead?

Reply:
I'm talking about code that has already been written and will break in subtle ways.

Reply:
I'll open a separate PR to fix abort